libtiledbsoma/src/soma/soma_array.h

```diff
@@ -185,6 +186,19 @@ class SOMAArray {
      */
     std::shared_ptr<Context> ctx();
 
+    std::optional<std::string> type() {
```

Comment:
```suggestion
    std::optional<std::string> soma_object_type() {
```

The word `type` is too overloaded already